products/BellHybrid/apps/application-bell-meditation-timer/models/ChimeVolume.cpp

```diff
@@ -8,11 +8,12 @@ namespace app::meditation::models
 
     void ChimeVolume::setValue(std::uint8_t value)
     {
+        defaultValue = value;
```

Comment:
what is the point of defaultValue if it is changed on each setVolume?
It will be always the same as value stored in model, right?

Reply:
Good point! From what I see in the settings application, it works because models are recreated each time a user enters the application. In this case model is created only once and it is loaded only once with the default value.

Reply:
It has to be done this way by "remembering" the previously set variable in a class member defaultValue, otherwise we have erroneous behavior when returning back without save (back press). Without storing defaultValue, e.g if we do just this: 

`    void ChimeVolume::restoreDefault()
    {
        setValue(audioModel.getVolume(AbstractAudioModel::PlaybackType::Meditation).value_or(0));
    }
`
If we change the volume from e.g. 6 to 7 and don't approve by light press, just use back press, upon re-entering Volume settings again we will see a value of 7 which is incorrect

Reply:
> Good point! From what I see in the settings application, it works because models are recreated each time a user enters the application. In this case model is created only once and it is loaded only once with the default value.

The fact that the model here is created only once implies that defaultValue variable has to be updated every time in setValue() since it won't get the proper updated value from the models constructor (as it is in the models in settings app). This implementation seems to work well on the hardware

Reply:
Update: It also works on the hardware :)